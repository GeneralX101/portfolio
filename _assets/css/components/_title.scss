.title {
  position: relative;
  margin: 0 0 85px 0;
  text-align: center;
  text-transform: capitalize;
  &:after {
    position: absolute;
    content: '';
    left: 50%;
    transform: translateX(-50%);
    bottom: -8px;
    width: 50px;
    height: 2px;
    background-color: $primary;
  }
}
