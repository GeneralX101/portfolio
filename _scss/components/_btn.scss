.btn + .btn {
  margin-top: 2em;
  @include breakpoint(350px) {
    margin-top: 0;
    margin-left: 2em;
  }
}
.btn, button, input[type="submit"],input[type="reset"],input[type="button"] {
    position: relative;
    display: inline-block;
    padding: 18px 30px;
    font-size: 11px;
    font-family: inherit;
    line-height: 1.5;
    letter-spacing: 0.2em;
    text-decoration: none;
    text-transform: uppercase;
    white-space: nowrap;
    cursor: pointer;
    color: $bg-color;
    background-color: $button-dark;
    text-align: center;
    border: none;
    transition: all 0.45s cubic-bezier(0.25, 1, 0.33, 1);
    &:after {
      display: none;
    }
    &:hover, :focus, :active {
      color: $bg-color;
      background-color: lighten($button-dark, 14%);
      outline: 0;
    }
}
.btn-lg {
  padding: 20px 48px;
  padding: 2rem 4.8rem;
  position: relative;
}

.btn-primary {
  background-color: $primary!important;
  color: $bg-color!important;
  &:hover, :focus, :active {
    color: $bg-color;
    background-color: $primary-light;
    outline: 0;
  }
}

.pill {
    position: relative;
    display: inline-block;
    padding: 15px 42px;
    font-size: 11px;
    letter-spacing: 0.2em;
    text-decoration: none;
    text-transform: capitalize;
    color: $bg-color;
    background-color: $button-dark;
    border-radius: 300px;
    line-height: 1.5;
    text-align: center;
    border: none;
    transition: all 0.45s cubic-bezier(0.25, 1, 0.33, 1);
    &:after {
      display: none;
    }
    &:hover, :focus, :active {
      color: $bg-color;
      background-color: $dark-light;
    }
    &:focus {
      outline: none;
    }
}
.pill-lg {
  padding: 20px 48px;
  padding: 2rem 4.8rem;
  position: relative;
}
.btn-wrap {
  text-align: center;
  @include breakpoint($md) {
    text-align: left;
  }
}
.btn-center {
  text-align: center;
}
button {
  &.loading {
    position: relative;
    background-color: $text-color;
    transition: background-color .2s ease;
  }
}

.spin {
  content: "";
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  height: 30px;
  width: 30px;
  border-radius: 50%;
  border-top: 3px solid $primary-light;
  border-right: 3px solid $primary-light;
  border-bottom: 3px solid $primary-light;
  border-left: 3px solid $primary;
  animation: load 1.1s infinite linear;
}

@keyframes load {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
