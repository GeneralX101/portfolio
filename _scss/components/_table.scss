table {
  display: block;
  width: 100%;
  overflow-x: auto;
  max-width: 100%;
  margin-bottom: 1rem;
  border-collapse: collapse;
  border-spacing: 0;
  border: 1px solid #e0e0e0;
  border-radius: 4px;
  th, td {
    padding: 6px 13px;
    border: 1px solid #dfe2e5;
  }
  tr {
    background-color: #fff;
    border-top: 1px solid #c6cbd1;
  }
}
