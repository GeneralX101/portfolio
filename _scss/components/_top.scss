#to-top {
  position: fixed;
  display: block;
  text-align: center;
  right: 1.8em;
  bottom: -29px;
  cursor: pointer;
  height: 35px;
  width: 35px;
  // border-radius: 50%;
  background-color: rgba(103, 103, 103, 0.65);
  visibility: hidden;
  opacity: 1;
  z-index: 2;
  transition: all 0.25s cubic-bezier(0.86, 0, 0.07, 1);
  &.show {
    opacity: 1;
    visibility: visible;
    bottom: 1.8em;
  }
  svg {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    width: 10.5px;
    height: auto;
    .st1 {
      fill: none;
      stroke: $bg-color;
      stroke-width:2;
      stroke-linecap:round;
      stroke-miterlimit:10;
    }
  }
}
