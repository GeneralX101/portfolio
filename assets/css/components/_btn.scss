.btn + .btn {
  margin-top: 2em;
  @include breakpoint($sm) {
    margin-top: 0;
    margin-left: 2em;
  }
}
.btn, button, input[type="submit"],input[type="reset"],input[type="button"] {
    position: relative;
    display: inline-block;
    padding: 18px 30px;
    font-size: 11px;
    line-height: 1.5;
    letter-spacing: 0.2em;
    text-decoration: none;
    text-transform: uppercase;
    white-space: nowrap;
    cursor: pointer;
    color: $bg-color;
    background-color: $button-dark;
    text-align: center;
    border: 1px solid transparent;
    transition: all 0.45s cubic-bezier(0.25, 1, 0.33, 1);
    &:after {
      display: none;
    }
    &:hover, :focus, :active {
      color: $bg-color;
      box-shadow: 0 40px 60px rgba(0,0,0,.15);
      outline: 0;
    }
}
.btn-lg {
  padding: 20px 48px;
  padding: 2rem 4.8rem;
  position: relative;
}

.pill, button {
    position: relative;
    display: inline-block;
    padding: 15px 42px;
    font-size: 11px;
    letter-spacing: 0.2em;
    text-decoration: none;
    text-transform: capitalize;
    color: $bg-color;
    background-color: $button-dark;
    border-radius: 300px;
    line-height: 1.5;
    text-align: center;
    border: 1px solid transparent;
    transition: all 0.45s cubic-bezier(0.25, 1, 0.33, 1);
    &:after {
      display: none;
    }
    &:hover, :focus, :active {
      color: $bg-color;
      background-color: $dark-light;
    }
    &:focus {
      outline: none;
    }
}
.pill-lg {
  padding: 20px 48px;
  padding: 2rem 4.8rem;
  position: relative;
}
