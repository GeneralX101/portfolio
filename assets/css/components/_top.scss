#top {
  position: fixed;
  display: block;
  text-align: center;
  right: 20px;
  bottom: -30px;
}
