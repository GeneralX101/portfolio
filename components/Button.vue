<template>
  <component
    :is="tag"
    :href="href"
    :to="to"
    :disabled="disabled"
    :type="type"
    :target="href ? '_blank' : null"
    :rel="href ? 'noopener' : null"
    :class="['btn', `btn--${color}`, {'btn--block': block}, {'btn--loading': loading}]"
    v-bind="$attrs"
    v-on="$listeners">
    <slot></slot>
  </component>
</template>

<script>
export default {
  props: {
    to: { type: [ String, Object ], default: null },
    href: { type: [ String, Object ] },
    block: { type: [Boolean], deafult: false },
    disabled: { type: Boolean, default: false },
    color: { type: String, default: 'primary' },
    loading: { type: Boolean, default: false },
  },
  computed: {
    tag() {
      if ( this.to && /https?/.test(this.to) === false ) {
        return 'nuxt-link'
      } else if ( this.href || /https?/.test(this.to) ) {
        return 'a'
      } else {
        return 'button'
      }
    },
    type() {
      return !this.to && !this.href ? 'button' : null;
    }
  }
}
</script>


<style lang="scss" scoped>
@import "~/assets/_vars.scss";

.btn {
  position: relative;
  display: inline-block;
  @extend %btn-reset;
  padding: .75rem 2rem;
  text-align: center;
  white-space: nowrap;
  line-height: 1.5;
  color: $white;
  font-weight: 500;
  border-radius: 200px;
  user-select: none;
  transition: all .2s cubic-bezier(.02,.01,.47,1);

  &::before {
    content: "";
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background: currentColor;
    border-radius: 50px;
    z-index: -1;
    opacity: 0;
    transform: scale(.2);
    transition: all .2s cubic-bezier(.02,.01,.47,1);
  }

  &:not(:disabled) {
    cursor: pointer;

    @include breakpoint($md) {
      &:hover {
      transition: all .3s cubic-bezier(.02,.01,.47,1);
      transform: scale(1.1);
      box-shadow: 0 1px 8px rgba(58,51,53,.4);

        &::before {
          opacity: .15;
          transition: all .3s cubic-bezier(.02,.01,.47,1);
          transform: scale(1);
        }
      }
    }
  }

  &:disabled {
    opacity: 0.8;
    cursor: not-allowed;
  }

  &--block {
    display: block;
    width: 100%;
  }

  &--primary {
    background-color: $primary;

  }

  &--success {
    background-color: $success;
  }

  &--error {
    background-color: $error;
  }
}
</style>
