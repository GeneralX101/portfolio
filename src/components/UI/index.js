// import Alert from './Alert';
import Box from './Box';
import Button from './Button';
import Emoji from './Emoji';
import Flex from './Flex';
import FormGroup from './FormGroup';
import FormErrorMsg from './FormErrorMsg';
import * as Grid from './Grid';
import Heading from './Heading';
import IconButton from './IconButton';
import Input from './Input';
import Label from './Label';
import Link from './Link';
import Modal from './Modal';
// import Spinner from './Spinner';
import Text from './Text';

const { Container, Row, Col } = Grid;

export {
  // Alert,
  Button,
  Box,
  Col,
  Container,
  Emoji,
  Flex,
  FormGroup,
  FormErrorMsg,
  IconButton,
  Input,
  Label,
  Link,
  Modal,
  Row,
  // Spinner,
  Text,
  Heading,
};
