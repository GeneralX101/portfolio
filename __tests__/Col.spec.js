import { create } from 'react-test-renderer';
import { Col } from '../src/components/UI';

describe('<Col />', () => {
  it('renders correctly with default styles', () => {
    const container = create(<Col />);
    expect(container).toMatchSnapshot();
  });
});
