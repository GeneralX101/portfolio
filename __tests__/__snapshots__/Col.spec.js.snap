// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Col /> renders correctly with default styles 1`] = `
.emotion-0 {
  display: block;
  width: 100%;
  min-height: 1px;
  padding-left: 16px;
  padding-right: 16px;
}

<div
  className="emotion-0"
/>
`;
